### ISSUE TITLE
python-docker vs python3-docker

### ISSUE THREAD

**Issue by thedanbob** [spantaleev/matrix-docker-ansible-deploy#356]
I started getting a deprecation notice in ansible informing me of this: https://docs.ansible.com/ansible/latest/reference_appendices/interpreter_discovery.html

Basically, the way ansible decides which python to run on the server will change in 2.12. When I tested it on my ubuntu server, ansible started running python 3 rather than python 2.

The only thing I've found that breaks is here: https://github.com/spantaleev/matrix-docker-ansible-deploy/blob/master/roles/matrix-base/tasks/server_base/setup_debian.yml#L31
When ansible runs python 3 this needs to be "python3-docker". Unfortunately, I can't find any simple and backwards-compatible way to detect the python version ansible is running on the server. There's [this ansible module](https://docs.ansible.com/ansible/latest/modules/python_requirements_info_module.html) but it only became available in ansible 2.7.

Perhaps the safest way to deal with it is to just install both `python-docker` and `python3-docker`?

**Comment by SuperSandro2000** [comment]
> Perhaps the safest way to deal with it is to just install both python-docker and python3-docker?

Probably not as it would pull in python2 which I don't want and which probably fails in the next Debian stable and the next Ubuntu release or the one after that.

**Comment by thedanbob** [comment]
Do you mean `ansible_playbook_python`? In any case that's not a safe assumption unless ansible is running on the server itself. For example, if the client is a mac running python from homebrew the path is going to be different than any linux server.

It's a pretty safe bet that ansible will lean toward python 3 from now on, so I suppose we could set the new default for interpreter discovery, assume that's going to be python 3 for 99% of people, change that one package, and cross our fingers.

**Comment by SuperSandro2000** [comment]
Yeah, google probably lead me to a stone age old ansible doc.

What is with ``ansible_python_version``? See https://docs.ansible.com/ansible/latest/user_guide/playbooks_variables.html#variables-discovered-from-systems-facts

**Comment by thedanbob** [comment]
Aha! That's exactly what I was looking for. I knew there must be a way to get the version of python on the server from within ansible but none of my searches turned up that page.

This works:
```yaml
- name: Ensure APT packages are installed
  apt:
    name:
      - "python{{'3' if ansible_python.version.major == 3}}-docker"
```
I'll submit a pull request.

**Comment by prasket** [comment]
Just a thought but since Python 2.7 was EOL on Jan 1, 2020 and will no longer receive security updates wouldn't it be better to convert everything in this project to using Python3?

**Comment by thedanbob** [comment]
Ansible already runs on both 2 and 3 so this should be the only place where it matters. Plus there doesn't seem to be any way to force ansible to run python3 besides explicitly setting the path to the binary, which obviously isn't viable.

**Comment by spantaleev** [comment]
Should be fixed by #358. Thank you for catching and fixing this! :+1:
